aspnetcore/signalr/authn-and-authz.md

```diff
@@ -0,0 +1,114 @@
+
+### Cookie Authentication
+
+In a simple web application, cookie authentication allows your existing user credentials to automatically flow to SignalR connections. When using the browser client, no additional configuration is needed. If the user is logged in to your application, the SignalR connection will automatically inherit this authentication.
+
+We do not recommend using Cookie authentication unless you only need to authenticate users from the browser client. When using the .NET Client, the `Cookies` property can be configured in the `.WithUrl` call in order to provide a cookie. However, this requires that you provide an API to exchange authentication data for a Cookie.
+
+### Bearer Token Authentication
+
+When using the .NET Client, or when your SignalR Hubs are located on a different server from your web application, we recommend using bearer token authentication. In this authentication scheme, you configure the SignalR client with an access token to send to the server. The server validates this token and uses the data within it to identify the user. The details of bearer token authentication are beyond the scope of this document, but there are some things to note when using this form of authentication in SignalR. On the server, bearer token authentication is configured using the [JWT Bearer middleware](/dotnet/api/microsoft.extensions.dependencyinjection.jwtbearerextensions.addjwtbearer?view=aspnetcore-2.1).
+
+In the JavaScript client, the token can be provided using the `accessTokenFactory` option.
+
+[!code-javascript[Configure Access Token](authn-and-authz/sample/wwwroot/js/chat.ts?range=63-65)]
+
+In the .NET client, there is a simlar `AccessTokenProvider` property that can be used to configure this:
+
+```csharp
+var connection = new HubConnectionBuilder()
+    .WithUrl("https://example.com/myhub", options => options.AccessTokenProvider = () => _myAccessToken)
+    .Build();
+```
+
+> [!NOTE]
+> This function is called before **every** HTTP request made by SignalR. This means that if you need to renew the token in order to keep the connection active (because it may expire during the connection) you can do so from within this function and SignalR will automatically start using the new token.
+
+In standard Web APIs, bearer tokens are sent via an HTTP header. However, due to security limitations imposed by browsers, the browser client is not able to set these headers when using the WebSockets or Server-Sent Events transports. When using those transports, the token is transmitted as a query string parameter. In order to support this on the server, additional configuration is required:
```

Comment:
via -> through or in 
is not -> isn't

Reply:
Ah man, I had it in my head that "via" was a word to avoid and I still threw one in here 😄 . My mind must just work via "via" all the time 😛

Reply:
That one took me a while too.